### ISSUE TITLE
Backend doesn't work on Python 3.11

### ISSUE THREAD

**Issue by nanaian** [decompme/decomp.me#636]
```
alex@mba decomp.me backend $ poetry run python3 manage.py runserver                                                                          main
21:00:54 INFO Watching for file changes with StatReloader
Performing system checks...

21:00:54 INFO Enabled 3 compiler(s): dummy, gcc2.7.2kmc, gcc2.8.1
21:00:54 INFO Available platform(s): dummy, n64
21:00:54 INFO Startup time: 1671969654
Exception in thread django-main-thread:
Traceback (most recent call last):
  File "/opt/homebrew/Cellar/python@3.11/3.11.0/Frameworks/Python.framework/Versions/3.11/lib/python3.11/threading.py", line 1038, in _bootstrap_inner
    self.run()
  File "/opt/homebrew/Cellar/python@3.11/3.11.0/Frameworks/Python.framework/Versions/3.11/lib/python3.11/threading.py", line 975, in run
    self._target(*self._args, **self._kwargs)
  File "/Users/alex/Library/Caches/pypoetry/virtualenvs/backend-x2beI-mP-py3.11/lib/python3.11/site-packages/django/utils/autoreload.py", line 64, in wrapper
    fn(*args, **kwargs)
  File "/Users/alex/Library/Caches/pypoetry/virtualenvs/backend-x2beI-mP-py3.11/lib/python3.11/site-packages/django/core/management/commands/runserver.py", line 134, in inner_run
    self.check(display_num_errors=True)
  File "/Users/alex/Library/Caches/pypoetry/virtualenvs/backend-x2beI-mP-py3.11/lib/python3.11/site-packages/django/core/management/base.py", line 475, in check
    all_issues = checks.run_checks(
                 ^^^^^^^^^^^^^^^^^^
  File "/Users/alex/Library/Caches/pypoetry/virtualenvs/backend-x2beI-mP-py3.11/lib/python3.11/site-packages/django/core/checks/registry.py", line 88, in run_checks
    new_errors = check(app_configs=app_configs, databases=databases)
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/alex/Library/Caches/pypoetry/virtualenvs/backend-x2beI-mP-py3.11/lib/python3.11/site-packages/django/core/checks/urls.py", line 42, in check_url_namespaces_unique
    all_namespaces = _load_all_namespaces(resolver)
                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/alex/Library/Caches/pypoetry/virtualenvs/backend-x2beI-mP-py3.11/lib/python3.11/site-packages/django/core/checks/urls.py", line 61, in _load_all_namespaces
    url_patterns = getattr(resolver, "url_patterns", [])
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/alex/Library/Caches/pypoetry/virtualenvs/backend-x2beI-mP-py3.11/lib/python3.11/site-packages/django/utils/functional.py", line 57, in __get__
    res = instance.__dict__[self.name] = self.func(instance)
                                         ^^^^^^^^^^^^^^^^^^^
  File "/Users/alex/Library/Caches/pypoetry/virtualenvs/backend-x2beI-mP-py3.11/lib/python3.11/site-packages/django/urls/resolvers.py", line 715, in url_patterns
    patterns = getattr(self.urlconf_module, "urlpatterns", self.urlconf_module)
                       ^^^^^^^^^^^^^^^^^^^
  File "/Users/alex/Library/Caches/pypoetry/virtualenvs/backend-x2beI-mP-py3.11/lib/python3.11/site-packages/django/utils/functional.py", line 57, in __get__
    res = instance.__dict__[self.name] = self.func(instance)
                                         ^^^^^^^^^^^^^^^^^^^
  File "/Users/alex/Library/Caches/pypoetry/virtualenvs/backend-x2beI-mP-py3.11/lib/python3.11/site-packages/django/urls/resolvers.py", line 708, in urlconf_module
    return import_module(self.urlconf_name)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/python@3.11/3.11.0/Frameworks/Python.framework/Versions/3.11/lib/python3.11/importlib/__init__.py", line 126, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "<frozen importlib._bootstrap>", line 1206, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1178, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1149, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 690, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 940, in exec_module
  File "<frozen importlib._bootstrap>", line 241, in _call_with_frames_removed
  File "/Users/alex/repos/decomp.me/backend/decompme/urls.py", line 5, in <module>
    path("api/", include("coreapp.urls")),
                 ^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/alex/Library/Caches/pypoetry/virtualenvs/backend-x2beI-mP-py3.11/lib/python3.11/site-packages/django/urls/conf.py", line 38, in include
    urlconf_module = import_module(urlconf_module)
                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/python@3.11/3.11.0/Frameworks/Python.framework/Versions/3.11/lib/python3.11/importlib/__init__.py", line 126, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "<frozen importlib._bootstrap>", line 1206, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1178, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1149, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 690, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 940, in exec_module
  File "<frozen importlib._bootstrap>", line 241, in _call_with_frames_removed
  File "/Users/alex/repos/decomp.me/backend/coreapp/urls.py", line 3, in <module>
    from coreapp.views import compilers, stats, project, scratch, user
  File "/Users/alex/repos/decomp.me/backend/coreapp/views/scratch.py", line 22, in <module>
    from ..decompiler_wrapper import DecompilerWrapper
  File "/Users/alex/repos/decomp.me/backend/coreapp/decompiler_wrapper.py", line 7, in <module>
    from coreapp.m2c_wrapper import M2CError, M2CWrapper
  File "/Users/alex/repos/decomp.me/backend/coreapp/m2c_wrapper.py", line 5, in <module>
    from m2c.src.main import parse_flags, run
  File "/Users/alex/repos/decomp.me/backend/m2c/src/main.py", line 11, in <module>
    from .flow_graph import FlowGraph, build_flowgraph, visualize_flowgraph
  File "/Users/alex/repos/decomp.me/backend/m2c/src/flow_graph.py", line 21, in <module>
    from .options import Formatter, Options, Target
  File "/Users/alex/repos/decomp.me/backend/m2c/src/options.py", line 165, in <module>
    @dataclass
     ^^^^^^^^^
  File "/opt/homebrew/Cellar/python@3.11/3.11.0/Frameworks/Python.framework/Versions/3.11/lib/python3.11/dataclasses.py", line 1221, in dataclass
    return wrap(cls)
           ^^^^^^^^^
  File "/opt/homebrew/Cellar/python@3.11/3.11.0/Frameworks/Python.framework/Versions/3.11/lib/python3.11/dataclasses.py", line 1211, in wrap
    return _process_class(cls, init, repr, eq, order, unsafe_hash,
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/python@3.11/3.11.0/Frameworks/Python.framework/Versions/3.11/lib/python3.11/dataclasses.py", line 959, in _process_class
    cls_fields.append(_get_field(cls, name, type, kw_only))
                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/python@3.11/3.11.0/Frameworks/Python.framework/Versions/3.11/lib/python3.11/dataclasses.py", line 816, in _get_field
    raise ValueError(f'mutable default {type(f.default)} for field '
ValueError: mutable default <class 'm2c.src.options.CodingStyle'> for field coding_style is not allowed: use default_factory
```

**Comment by nanaian** [comment]
Looks like this may need to be an issue we fix upstream on m2c

**Comment by simonlindholm** [comment]
does it work if you make the CodingStyle dataclass `frozen=True`?

**Comment by simonlindholm** [comment]
Yep, that seems to have worked

**Comment by nanaian** [comment]
Fixed in https://github.com/matt-kempster/m2c/commit/7103c2d4776dc64039a74900938f05875818bffa, yet to be merged into decomp.me though

**Comment by nanaian** [comment]
Fixed in #624
